### PR TITLE
[Dubbo-2335] Need to publish artifacts to snapshot repository if CI passes

### PR DESCRIPTION
**Pull request by chickenlj**
## What is the purpose of the change
See comments on #2335 
Automatically publishing artifacts to snapshot repository after CI.

## Brief changelog

travis.yml

## Verifying this change

check snapshot repository after a successful CI.

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/incubator-dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [x] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/incubator-dubbo/tree/master/dubbo-test).
- [x] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [x] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -16,6 +16,12 @@ script:
 
 after_success:
   - bash <(curl -s https://codecov.io/bash)
-
+  - "[[ $TRAVIS_BRANCH == 'master' ]] && [[ ${TRAVIS_PULL_REQUEST} == 'false' ]] &&
+    [[ ${TRAVIS_TAG} == '' ]] && mvn clean deploy --settings .travis.settings.xml"
 after_failure:
-  - if [ -f dubbo.log ]; then echo "------TAIL of dubbo.log------"; tail -n 1000 dubbo.log; echo "------END of dubbo.log------"; fi
+  - if [ -f dubbo.log ]; then echo "------TAIL of dubbo.log------"; tail -n 1000 dubbo.log;
+    echo "------END of dubbo.log------"; fi
```
Comment (htynkn) [inline .travis.yml]: what is `dubbo.log`? maven logs?

**Comment by htynkn** [comment]
I think `deploy` in travis CI is more suitable for this scenairo. We can use script as provider (refer to:https://docs.travis-ci.com/user/deployment/script/)

And `deploy` provider branch, repo, tag and other condition configuration, it looks more clear than check env like `BRANCH_NAME` in script

File: .travis.yml (hunk 1 above)
Comment (chickenlj) [reply .travis.yml]: I don't know the purpose of `dubbo.log` for sure.

I think maybe all maven logs are stored in this file.

**Comment by chickenlj** [comment]
> I think `deploy` in travis CI is more suitable for this scenairo. We can use script as provider (refer to:https://docs.travis-ci.com/user/deployment/script/)

I also find the `deploy` command, as I mentioned in the #2335. I tend to switch to `deploy` the next time, this time let's make sure the whole process works first.  Whant do you think? @htynkn

**Comment by htynkn** [comment]
@chickenlj yeah. let's make it works first.

**Comment by ralf0131** [comment]
I doubt this approach might violate ASF policy (not sure) since you place your Apache credentials on third party servers. If Travis is compromised your Apache credentials might be exposed. 

I think we should discuss it on the mailing list first.

**Comment by htynkn** [comment]
Considering ASF policy and also we have implement similar stuff apache jenkins (https://builds.apache.org/job/Apache%20Dubbo/job/apache-dubbo-snapshot-deployment/).

I'm going to close this PR.

**Pull request closed without merging**